beater/handlers.go

```diff
@@ -243,13 +271,41 @@ func sourcemapHandler(p processor.Processor, beaterConfig *Config, report report
 				processRequestHandler(p, conf.Config{SmapMapper: smapper}, report, decoder.DecodeSourcemapFormData))))
 }
 
+// Deprecated: use rootHandler instead
 func healthCheckHandler() http.Handler {
 	return logHandler(
 		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			sendStatus(w, r, okResponse)
 		}))
 }
 
+func rootHandler(secretToken string) http.Handler {
+	serverInfo := common.MapStr{
+		"build_date": version.BuildTime().Format(time.RFC3339),
+		"build_sha":  version.Commit(),
+		"version":    version.GetDefaultVersion(),
+	}
+	detailedOkResponse := serverResponse{
+		code:    http.StatusOK,
+		counter: responseOk,
+		body:    serverInfo,
+	}
+
+	var handler http.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path != "/" {
```

Comment:
is this actually necessary?

Reply:
Yes, I forgot about how the [default mux](https://golang.org/pkg/net/http/#ServeMux.Handle) pattern matches work until writing https://github.com/elastic/apm-server/pull/1197/files#diff-24302485aab47c0697eb5cb545fbe394R120.